prov/shm/src/smr_progress.c

```diff
@@ -492,9 +492,9 @@ static int smr_progress_ipc(struct smr_cmd *cmd, enum fi_hmem_iface iface,
 				(void **) &cmd->msg.data.ipc_info.fd_handle,
 				&ipc_fd, ipc_device, &base);
 	} else {
-		ret = ofi_hmem_open_handle(cmd->msg.data.ipc_info.iface,
-				(void **) &cmd->msg.data.ipc_info.ipc_handle,
-				device, &base);
+		ret = ipc_cache_map_memhandle(ep->hmem_cache,
+									  &cmd->msg.data.ipc_info,
+									  &base);
```

Comment:
In subsequent calls the cache is queried and the mapped address is retrieved. But in L520 the handle is already closed so the address is invalid. Do I miss any change that fixes this issue?

Reply:
Thank you for your work! I am actually testing your commits in my development for using CUDA IPC in shm, I will leave more comments when I have some more issues.